### PR TITLE
Add JS linting to CI

### PR DESCRIPTION
**Pull request by alistairjcbrown**
Adds the JS linting task to CI script

### CODE REVIEW

File: docker-compose-dev.yml [hunk 1]
```diff
@@ -87,12 +87,15 @@ services:
       - '3007:3000' # expose ports - HOST:CONTAINER
     environment:
       - NODE_ENV=development
+      - CI=${CI}
```
Comment (apoclyps) [inline docker-compose-dev.yml]: This will also need to be added to the staging/production docker-compose.

File: test.sh [hunk 2]
```diff
@@ -4,10 +4,10 @@ env=$1
 file=""
 fails=""
 
-if [[ "${env}" == "stage" ]]; then
-  file="docker-compose-dev.yml"
-elif [[ "${env}" == "dev" ]]; then
+if [[ "${env}" == "dev" ]]; then
```
Comment (apoclyps) [inline test.sh]: The default `env` is always 'stage' - all tests run the file set as `docker-compose-stage.yml`: https://github.com/apoclyps/my-dev-space/blob/master/.travis.yml#L62

File: docker-compose-dev.yml (hunk 1 above)
Comment (alistairjcbrown) [reply docker-compose-dev.yml]: @apoclyps Thanks! I was wondering why that wasn't running on CI

File: test.sh (hunk 2 above)
Comment (alistairjcbrown) [reply test.sh]: @apoclyps Should stage be using `docker-compose-stage.yml`?

Comment (apoclyps) [reply test.sh]: yes; it should. Although right now there is very little difference between dev and stage docker compose files. The only difference was the dev Dockerfiles had volume mounts (which are not being used anymore). feel free to correct it in this PR or leave it as a follow up.

Comment (alistairjcbrown) [reply test.sh]: @apoclyps That's what I've aimed to do here, with the order dev -> stage -> prod 👍

**Comment by apoclyps** [comment]
I tried producing this locally; look like the time out's are happening outside of travis when `docker-compose -f docker-compose-stage.yml run client-test npm run lint` is run.

**Pull request merged**